### PR TITLE
Fix test in auto_trace_test.cc

### PR DESCRIPTION
**Pull request by Kozoriz**
Reason : 1. Randomly crashes

Changes: 1. Corrected logger.cc
	 2. Fixed test in auto_trace_test.cc 
	 3. Added methods to threads
	 4. Added specific config file

Related: APPLINK-19521

### CODE REVIEW

**Comment by Kozoriz** [comment]
please  review : @malirod, @AByzhynar , @VVeremjova , @OHerasym

File: src/components/include/utils/threads/message_loop_thread.h [hunk 1]
```diff
@@ -86,6 +86,9 @@ class MessageLoopThread {
   // Process already posted messages and stop thread processing. Thread-safe.
   void Shutdown();
 
+  // Added for utils/test/auto_trace_test.cc
+  size_t GetMessageQueueSize() const;
```
Comment (AByzhynar) [inline src/components/include/utils/threads/message_loop_thread.h]: Do you use this method somewhere except of unit-tests? 
If no - may be wrap into #ifdef BUILD_TESTS?

File: src/components/utils/src/threads/posix_thread.cc [hunk 2]
```diff
@@ -249,6 +249,10 @@ bool Thread::start(const ThreadOptions& options) {
   return pthread_result == EOK;
 }
 
+void Thread::yield() {
+  sched_yield();
```
Comment (AByzhynar) [inline src/components/utils/src/threads/posix_thread.cc]: Add #include <sched.h>

File: src/components/include/utils/threads/message_loop_thread.h (hunk 1 above)
Comment (pestOO) [inline src/components/include/utils/threads/message_loop_thread.h]: Let it be a part of public interface.
Using differ interfaces in and out of test is not good practise. 
But in this case we need to add an appropriate unit-test, don't we?

Comment (AByzhynar) [inline src/components/include/utils/threads/message_loop_thread.h]: Ok. From this point of view I agreed.

File: src/components/utils/src/threads/posix_thread.cc (hunk 2 above)
Comment (AByzhynar) [inline src/components/utils/src/threads/posix_thread.cc]: Also check #ifdef BUILD_TESTS wrapper at the beginning of file. Probably it is not required now.

File: src/components/utils/test/CMakeLists.txt [hunk 3]
```diff
@@ -86,9 +87,9 @@ else ()
...
   find_package(Sqlite3 REQUIRED)
   list (APPEND testSources
-    ./sqlite_wrapper/sql_database_test.cc 
-    ./sqlite_wrapper/sql_query_test.cc   
-    generated_code_with_sqlite_test.cc   
+    ./sqlite_wrapper/sql_database_test.cc
+    ./sqlite_wrapper/sql_query_test.cc
+    generated_code_with_sqlite_test.cc
 
     # TODO{ALeshin} APPLINK-11132 AssertTrue in SetUpTestCase() return false
```
Comment (AByzhynar) [inline src/components/utils/test/CMakeLists.txt]: Check this commented code. Task is implemented already. If it really ok - uncomment code and delete comment,

File: src/components/utils/test/auto_trace_test.cc [hunk 4]
```diff
@@ -86,17 +100,26 @@ void DeinitLogger() {
 }
 
 //TODO(VVeremjova) Logger does not write debug information in file
```
Comment (AByzhynar) [inline src/components/utils/test/auto_trace_test.cc]: Why do we need this comment?

File: src/components/include/utils/threads/message_loop_thread.h (hunk 1 above)
Comment (Kozoriz) [inline src/components/include/utils/threads/message_loop_thread.h]: @pestOO , @AByzhynar  see  	4779d04

**Comment by Kozoriz** [comment]
@AByzhynar see cfa95fd (amended)

**Comment by malirod** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Comment by OHerasym** [comment]
Reviewed

**Comment by Kozoriz** [comment]
assignee @VVeremjova

**Pull request merged**